src/NHibernate/Dialect/SybaseSQLAnywhere10Dialect.cs

```diff
@@ -192,6 +194,11 @@ protected virtual void RegisterAggregationFunctions()
 
 		protected virtual void RegisterBitFunctions()
 		{
+			// SQL Anywhere does not respect usual priorities with the bitwise not. Unfortunately the HQL parser
+			// furthermore remove "undue" parenthesis according to usual rules. As the bitwise not should have maximal
+			// priority, we can work around this by using a template forcing parenthesis around it.
+			RegisterFunction("bnot", new VarArgsSQLFunction(NHibernateUtil.Int64, "(~", "", ")"));
```

Comment:
This is not a breaking change, because the function argument will by default be all that follows the `~`. But thanks to this change, if parenthesis are used to explicit the priority, they will be taken into account for supplying the "function" argument, instead of being silently removed by the parser.